compiler/src/iree/compiler/InputConversion/MHLO/ConvertMHLOToLinalgExt.cpp

```diff
@@ -470,6 +471,125 @@ struct ReverseOpConversion : public OpConversionPattern<mhlo::ReverseOp> {
   }
 };
 
+//===----------------------------------------------------------------------===//
+// TopkOp
+//===----------------------------------------------------------------------===//
+
+struct TopkOpConversion : public OpConversionPattern<chlo::TopKOp> {
+  using OpConversionPattern<chlo::TopKOp>::OpConversionPattern;
+  LogicalResult matchAndRewrite(
+      chlo::TopKOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const final {
+    Location loc = op.getLoc();
+    Value operand = adaptor.operand();
+
+    auto inputValuesType = operand.getType().dyn_cast<RankedTensorType>();
+    if (!inputValuesType) {
+      return failure();
+    }
+    uint64_t kDim = inputValuesType.getRank() - 1;
+
+    // Create a range of indicies for the input tensor match input shape but
+    // with i32 type.
+    SmallVector<Value> dynSizes;
+    for (auto en : llvm::enumerate(inputValuesType.getShape())) {
+      if (en.value() == ShapedType::kDynamicSize) {
+        dynSizes.push_back(rewriter.create<tensor::DimOp>(
+            loc, adaptor.getOperands()[0], en.index()));
+      }
+    }
+    Value initAcc = rewriter.create<linalg::InitTensorOp>(
+        loc, dynSizes, inputValuesType.getShape(), rewriter.getI32Type());
+
+    // Create the indexing maps for the generic
+    size_t inputRank = inputValuesType.getRank();
+    AffineMap mapIdentity = rewriter.getMultiDimIdentityMap(inputRank);
+    SmallVector<AffineMap> indexingMaps{mapIdentity};
+    SmallVector<StringRef> iterators{inputRank, "parallel"};
```

Comment:
Prefer using "assignment-like syntax" or "traditional constructor syntax (with parentheses"

```suggestion
    SmallVector<AffineMap> indexingMaps = {mapIdentity};
    SmallVector<StringRef> iterators(inputRank, "parallel");
```